### ISSUE TITLE
Error evaluating 'query._query.canonicalId' when using useFirestoreCollectionData

### ISSUE THREAD

**Issue by jylin** [FirebaseExtended/reactfire#221]
### Version info

**React:** 16.13.0

**Firebase:** 7.10.0

**ReactFire:** 2.0.1

**Other (e.g. Node, browser, operating system) (if applicable):**

### Test case

The sample in the reactfire codebase seems to throw this exception without any changes.

### Steps to reproduce

Run the sample app included in this codebase.

### Expected behavior

No crash

### Actual behavior

```
TypeError: undefined is not an object (evaluating 'query._query.canonicalId')

getHashFromFirestoreQuery
node_modules/reactfire/firestore/index.js:33
  30 |     return useObservable(fromCollectionRef(query), queryId, checkStartWithValue(options));
  31 | }
  32 | function getHashFromFirestoreQuery(query) {
> 33 |     return query._query.canonicalId();
  34 | }
  35 | export function useFirestoreCollectionData(query, options) {
  36 |     var idField = checkIdField(options);
```

**Comment by jylin** [comment]
I reverted to firebase version 7.5.0, and that seems to fix it, so it's probably due to a change to firebase rather than in reactfire. Didn't try bisecting to see which exact version/change causes the issue though.

**Comment by 16oh4** [comment]
I'm having the same issue with Firebase 7.10.0 and the rest of these deps `"@material-ui/core": "^4.9.5",
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.3.2",
    "@testing-library/user-event": "^7.1.2",
    "firebase": "^7.10.0",
    "react": "^0.0.0-experimental-235a6c4af",
    "react-dom": "^0.0.0-experimental-235a6c4af",
    "react-scripts": "3.4.0",
    "reactfire": "^2.0.1"`

**Comment by pablogravielseo** [comment]
Same problem here, I was using version "7.9.3", now I've installed the version "v7.8.2" and is working.

**Comment by jamesdaniels** [comment]
I believe this is due to Firebase 7.9 mangling the private API that we rely on in CJS builds... thoughts @schmidt-sebastian?

**Comment by schmidt-sebastian** [comment]
Talked to @jamesdaniels, who will add an externs file for the private APIs that are used here and add it here: https://github.com/firebase/firebase-js-sdk/blob/master9/packages/firestore/externs.json

**Comment by jamesdaniels** [comment]
After further discussions we found a solution that does not rely on a private API. We'll have a work around out shortly.

**Comment by jamesdaniels** [comment]
Fixed in `2.0.2`
